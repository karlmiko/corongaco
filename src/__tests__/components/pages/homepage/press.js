import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import Press from '../../../../components/pages/homepage/press'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    allHomepagePressYaml: {
      edges: [
        {
          node: {
            name: 'test A',
            logo: 'test-a.png',
            width: 300,
            featured: false,
          },
        },
        {
          node: {
            name: 'test B',
            logo: 'test-b.png',
            width: 300,
            featured: true,
          },
        },
      ],
    },
    allCovidPress: {
      edges: [
        {
          node: {
            id: '5ce2bc4d-3d24-5e1a-91bb-a30d098ccfc7',
            title:
              'Coronavirus testin: California continues to lag compared to other states',
            url:
              'https://abc7news.com/coronavirus-testing-california-covid-19-in-tests-ca-pandemic/6111601/',
            publication: 'ABC 7 News',
            publishDate: 'April 18, 2020',
          },
        },
        {
          node: {
            id: '68c6166f-7498-5eaf-bd6a-178a5e13acf2',
            title:
              'Amid flurry of national coronavirus data, Massachusetts remains a hot spot of infection',
            url:
              'https://www.bostonglobe.com/2020/04/18/nation/amid-flurry-national-coronavirus-data-massachusetts-remains-hot-spot-infection/',
            publication: 'The Boston Globe',
            publishDate: 'April 18, 2020',
          },
        },
      ],
    },
  }))
})

describe('Components : Pages : Homepage : Press', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Press />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
