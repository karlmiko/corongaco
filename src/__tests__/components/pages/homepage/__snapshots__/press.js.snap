// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Homepage : Press renders correctly 1`] = `
<section>
  <h3
    className="header"
  >
    Who's using our data
  </h3>
  <div
    className="press"
  >
    <div
      className="list"
    >
      <ul
        className="press-list pressList"
      >
        <li>
          <cite
            className="title"
          >
            <a
              href="https://abc7news.com/coronavirus-testing-california-covid-19-in-tests-ca-pandemic/6111601/"
            >
              Coronavirus testin: California continues to lag compared to other states
            </a>
          </cite>
          <div
            className="detail-text detailText"
          >
            <span
              className="source"
            >
              ABC 7 News
            </span>
            <span
              aria-hidden={true}
              className="dotSeparator"
            >
              •
            </span>
            April 18, 2020
          </div>
        </li>
        <li>
          <cite
            className="title"
          >
            <a
              href="https://www.bostonglobe.com/2020/04/18/nation/amid-flurry-national-coronavirus-data-massachusetts-remains-hot-spot-infection/"
            >
              Amid flurry of national coronavirus data, Massachusetts remains a hot spot of infection
            </a>
          </cite>
          <div
            className="detail-text detailText"
          >
            <span
              className="source"
            >
              The Boston Globe
            </span>
            <span
              aria-hidden={true}
              className="dotSeparator"
            >
              •
            </span>
            April 18, 2020
          </div>
        </li>
      </ul>
    </div>
    <div
      className="logos"
    >
      <div
        className="homepage-press-logos logos undefined"
      >
        <p
          className="a11y-only"
        >
          News organizations including The New York Times, The Washington Post, CNN, Vox, ProPublica, and The Wall Street Journal use our data in their reporting.
        </p>
        <img
          alt="test B logo"
          aria-hidden={true}
          src="/images/press-logos/test-b.png"
        />
      </div>
    </div>
    <a
      className="cta centered "
      href="/about-project/in-the-press"
    >
      See what else our data powers
      <span
        aria-hidden={true}
        className="arrow"
      >
         
        →
      </span>
    </a>
  </div>
</section>
`;
